go-controller/pkg/kubevirt/router.go

```diff
@@ -166,33 +179,73 @@ func EnsureLocalZonePodAddressesToNodeRoute(watchFactory *factory.WatchFactory,
 		}); err != nil {
 			return fmt.Errorf("failed adding static route: %v", err)
 		}
+
 	}
 	return nil
 }
 
-// EnsureRemoteZonePodAddressesToNodeRoute will add static routes when live
-// migrated pod belongs to remote zone to send traffic over transwitch switch
-// port of the node where the pod is running:
-//   - A dst-ip with live migrated pod ip as prefix and nexthop the pod's
-//     current node transit switch port.
-func EnsureRemoteZonePodAddressesToNodeRoute(watchFactory *factory.WatchFactory, nbClient libovsdbclient.Client, pod *corev1.Pod, nadName string) error {
+func reconcileClusterDefaultNetworkLocalZoneNeighbours(watchFactory *factory.WatchFactory, pod *corev1.Pod, podAnnotation *util.PodAnnotation, nadName string) error {
+	// TODO: Implement neighbours update for non IC
+	if !config.OVNKubernetesFeature.EnableInterconnect {
+		return nil
+	}
+
+	currentNode := os.Getenv("K8S_NODE")
+	currentNodeOwnsSubnet, err := nodeContainsPodSubnet(watchFactory, currentNode, podAnnotation, nadName)
+	if err != nil {
+		return err
+	}
+
+	vmRunningAtCurrentNode := pod.Spec.NodeName == currentNode
+
+	// Now that the VM is running at a node not owning his subnet, using pod's mac
+	// will not work since it's a different switch so overwrite those neighbors with
+	// proxy mac from gatway
+	if !currentNodeOwnsSubnet && vmRunningAtCurrentNode {
```

Comment:
So on a node that does not own the subnet but has the VM migrated to we send GARPs for all IPs that are running on the original node?

Reply:
Right, since VM is at node that need to do routing to access IPs at the VM's subnet we need to update VM's neighbor cache so those IPs access over the gateway since they are not part of the node's switch.

If we don't do so, the traffic will be down until the ip neighors at VM invalidates and send a normal ARP that will be responded with the proxy ARP and traffic will work, but that can take at to 10 seconds or so from what I have see, and depends on guest VM network configuration.